src/storage.ts

```diff
@@ -15,3 +17,26 @@ export async function setPersistedProjects(projects: Project[]) {
     projects: JSON.stringify(projects),
   });
 }
+
+export async function getPersistedEntries(): Promise<Entry[]> {
+  const temp = await chrome.storage.sync.get(null);
+  const result = await chrome.storage.sync.get("entries");
+  if (result.entries !== undefined) {
+    return JSON.parse(result.entries);
+  } else {
+    return [];
+  }
+}
+
+export async function setPersistedEntries(entries: Entry[]) {
+  await chrome.storage.sync.set({
+    entries: JSON.stringify(entries),
+  });
+}
+
+export async function addPersistedEntry(entry: Entry) {
+  await getPersistedEntries().then((entries) => {
```

Comment:
We don’t need the `then` if we use `await`:
```javascript
const entries = await getPersistedEntries();
entries.push(entry);
setPersistedEntries(entries);
```

Reply:
lol oops